### ISSUE TITLE
How to train the models

### ISSUE THREAD

**Issue by Wangman1** [lhwcv/mlsd_pytorch#2]
Hi, thank you for your code, and I want to know how to train the model in your code? thank you!

**Comment by lhwcv** [comment]
Sorry for my late reply.  (A little busy at work)
My result is  lower than in the paper,  about ~2-3  percent lower than  it.   Maybe next week I'll push my training code.

**Comment by Wangman1** [comment]
Thank you for your reply.
Looking forward to your open source！




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "lhwcv/mlsd_pytorch"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年7月13日(星期二) 中午11:57
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [lhwcv/mlsd_pytorch] How to train the models (#2)





 
Sorry for my late reply.  (A little busy at work)
 My result is  lower than in the paper,  about ~2-3  percent lower than  it.   Maybe next week I'll push my training code.

**Comment by lhwcv** [comment]
@Wangman1  Here is my result in mlsd_pytorch/README.md

**Comment by lhwcv** [comment]
<html><body>


model | img_size | sAP10
-- | -- | --
mlsd_tiny (this repo) | 512 | 56.4
mlsd_tiny (in the paper) | 512 | 58.0
mlsd_large (this repo) | 512 | 59.6
mlsd_large (in the paper) | 512 | 62.1


</body>
</html>

**Comment by Wangman1** [comment]
Wow, these results look great, and I can't wait to see how you train the model,  so great job, thank you!!!

**Comment by Wangman1** [comment]
Hi, when I train the model in your mode, raise the follow error:
![image](https://user-images.githubusercontent.com/33320872/126311901-a34e9932-a19d-4f1d-8f65-a5ec845e5195.png)

**Comment by Wangman1** [comment]
And the Google link to JSON file cannot download the content
![image](https://user-images.githubusercontent.com/33320872/126320682-701fdca6-ffb7-41e7-a861-f26507d292ee.png)

**Comment by lhwcv** [comment]
@Wangman1   
1.  pip install yacs
2.  use https://github.com/lhwcv/mlsd_pytorch/blob/main/data/wireframe_anno.zip
3. google cloud disk I'll check the issue later

**Comment by Wangman1** [comment]
@lhwcv 
Thank you so much for your help. My code has been successfully run!!!
![image](https://user-images.githubusercontent.com/33320872/126426744-0f3a9ca9-6030-4b22-82c2-e77334048085.png)

**Comment by YangTao516** [comment]
> @lhwcv Thank you so much for your help. My code has been successfully run!!! ![image](https://user-images.githubusercontent.com/33320872/126426744-0f3a9ca9-6030-4b22-82c2-e77334048085.png)

hi, could you please share the datasets with me ?  It seems that the url for the datasets is unusable.

**Comment by harolddu** [comment]
> model	img_size	sAP10
> mlsd_tiny (this repo)	512	56.4
> mlsd_tiny (in the paper)	512	58.0
> mlsd_large (this repo)	512	59.6
> mlsd_large (in the paper)	512	62.1



Thanks for your code and I can run the demo successfully. But loss values are both "nan" on wireframe and custom datasets during training, do you have any idea?

**Comment by michelebechini** [comment]
Hi, how long does it take (on average obviously) to train this model? I am trying using Colab GPUs and the process seems extremely slow...

**Comment by chenenp** [comment]
the Google link to JSON file cannot download the content.,please help me !
